XVim2/Xcode/SourceEditorViewProxy+XVim.m

```diff
@@ -131,8 +131,19 @@ - (void)xvim_endEditTransaction
     }
 }
 
+/// XVim last set insertionPoint is wrong when Xcode trim whitespace.
+/// If insertionPoint is wrong, wired position cursor jump will occur when `j` and `o` operation.
+/// This method fixing insertionPoint for that operation.
+- (void)xvim_fixInsertionPoint
+{
+    if (self.selectionMode == XVIM_VISUAL_NONE && self.insertionPoint > self.selectedRange.location) {
```

Comment:
if selectionMode condition is nothing, `j` will not move cursor in visual line mode when current cursor is blank line.